# Import Libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

## Load data

In [2]:
df=pd.read_csv("netflix.csv")

In [3]:
df.head() # show first five rows of data

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

## Check the shape of data

In [4]:
df.shape

(8807, 12)

**Observation**
- There is **8807** rows and **12** columns 

## Check the Data type of data

In [5]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

In [6]:
df.sort_values(by='release_year',ascending=False)

show_id     type                                          title  \
693     s694    Movie                         Ali & Ratu Ratu Queens   
781     s782    Movie          Black Holes | The Edge of All We Know   
762     s763    Movie                                   Sweet & Sour   
763     s764  TV Show                                    Sweet Tooth   
764     s765    Movie                   Trippin' with the Kandasamys   
...      ...      ...                                            ...   
8660   s8661    Movie  Undercover: How to Operate Behind Enemy Lines   
8763   s8764    Movie                WWII: Report from the Aleutians   
8205   s8206    Movie                           The Battle of Midway   
7790   s7791    Movie                                 Prelude to War   
4250   s4251  TV Show              Pioneers: First Women Filmmakers*   

            director                                               cast  \
693   Lucky Kuswandi  Iqbaal Ramadhan, Nirina Zubir, Asri Welas, Tik...   
781    Peter Galison                                                NaN   
762    Lee Kae-byeok          Jang Ki-yong, Chae Soo-bin, Jung Soo-jung   
763              NaN  Nonso Anozie, Christian Convery, Adeel Akhtar,...   
764    Jayan Moodley  Jailoshini Naidoo, Maeshni Naicker, Madhushan ...   
...              ...                                                ...   
8660       John Ford                                                NaN   
8763     John Huston                                                NaN   
8205       John Ford                          Henry Fonda, Jane Darwell   
7790     Frank Capra                                                NaN   
4250             NaN                                                NaN   

            country         date_added  release_year rating  duration  \
693             NaN      June 17, 2021          2021  TV-14   101 min   
781             NaN       June 2, 2021          2021  TV-14    99 min   
762     South Korea       June 4, 2021          2021  TV-14   103 min   
763   United States       June 4, 2021          2021  TV-14  1 Season   
764    South Africa       June 4, 2021          2021  TV-14    94 min   
...             ...                ...           ...    ...       ...   
8660  United States     March 31, 2017          1943  TV-PG    61 min   
8763  United States     March 31, 2017          1943  TV-PG    45 min   
8205  United States     March 31, 2017          1942  TV-14    18 min   
7790  United States     March 31, 2017          1942  TV-14    52 min   
4250            NaN  December 30, 2018          1925  TV-14  1 Season   

                                              listed_in  \
693              Comedies, Dramas, International Movies   
781                                       Documentaries   
762     Comedies, International Movies, Romantic Movies   
763   TV Action & Adventure, TV Dramas, TV Sci-Fi & ...   
764     Comedies, International Movies, Romantic Movies   
...                                                 ...   
8660                      Classic Movies, Documentaries   
8763                                      Documentaries   
8205                      Classic Movies, Documentaries   
7790                      Classic Movies, Documentaries   
4250                                           TV Shows   

                                            description  
693   After his father's passing, a teenager sets ou...  
781   Follow scientists on their quest to understand...  
762   Faced with real-world opportunities and challe...  
763   On a perilous adventure across a post-apocalyp...  
764   To rekindle their marriages, best friends-turn...  
...                                                 ...  
8660  This World War II-era training film dramatizes...  
8763  Filmmaker John Huston narrates this Oscar-nomi...  
8205  Director John Ford captures combat footage of ...  
7790  Frank Capra's documentary chronicles the rise ...  
4250  This coll

In [7]:
df

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]